frontend/src/components/ui/DatePicker.tsx

```diff
@@ -0,0 +1,129 @@
+import { useId, useState, useEffect } from 'react';
+import { format, isValid, parse } from 'date-fns';
+import { ko } from 'date-fns/locale';
+import { CalendarDays } from 'lucide-react';
+import { Button, Calendar, Input } from '@freemed-kit/ui';
+import { Popover, PopoverTrigger, PopoverContent } from '@freemed-kit/ui';
+import { cn } from '../../utils/cn';
+
+interface DatePickerProps {
+  date?: Date;
+}
+
+export function DatePicker({ date }: DatePickerProps) {
+  const popoverId = useId();
+
+  const [selectedDate, setSelectedDate] = useState<Date | undefined>(undefined);
+  const [inputValue, setInputValue] = useState('');
+  const [month, setMonth] = useState(date);
+
+  useEffect(() => {
+    if (date) {
+      setSelectedDate(date);
+      setInputValue(format(date, 'yyyy.MM.dd', { locale: ko })); // 한국어 형식 적용
+      setMonth(date);
+    }
+  }, [date]);
+
+  const handleDayPickerSelect = (date: Date | undefined) => {
+    if (!date) {
+      setInputValue('');
+      setSelectedDate(undefined);
+    } else {
+      setSelectedDate(date);
+      setInputValue(format(date, 'yyyy.MM.dd', { locale: ko })); // 한국어 형식 적용
+    }
+  };
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value;
+    const caretPosition = e.target.selectionStart || 0;
+    const onlyNum = value.replace(/[^0-9]/g, '');
+
+    let formattedValue = '';
+    let newCaretPosition = caretPosition;
+
+    if (onlyNum.length <= 6) {
+      formattedValue = onlyNum.replace(/([0-9]{4})([0-9]{1,2})/, '$1.$2');
+    } else if (onlyNum.length <= 8) {
+      formattedValue = onlyNum.replace(
+        /([0-9]{4})([0-9]{2})([0-9]{1,2})/,
+        '$1.$2.$3',
+      );
+    } else {
+      formattedValue = value;
+    }
+
+    if (formattedValue[caretPosition - 1] === '.') {
+      newCaretPosition += 1;
+    } else if (value[caretPosition - 1] === '.') {
+      newCaretPosition -= 1;
+    }
+
+    setInputValue(formattedValue);
+
+    const parsedDate = parse(formattedValue, 'yyyy.MM.dd', new Date());
+    if (isValid(parsedDate)) {
+      setSelectedDate(parsedDate);
+      setMonth(parsedDate);
+    } else {
+      setSelectedDate(undefined);
+    }
+
+    requestAnimationFrame(() => {
+      e.target.setSelectionRange(newCaretPosition, newCaretPosition);
+    });
+  };
+
+  return (
+    <div>
+      <div
+        className={cn(
+          'relative w-[276px] text-left font-normal',
+          !selectedDate && 'text-muted-foreground',
+        )}
+      >
```

Comment:
```suggestion
      <div
        className={cn(
          'relative text-left font-normal',
          !selectedDate && 'text-muted-foreground',
        )}
      >
```
요기 불필요한 div 태그 없애고 width 지정한 이유 크게 없다면 지워도 될 것 같아요! (다른 줄에서도 `w-[276px]` 전부) 그러면 반응형에 맞게 적용돼요!